be/src/storage/lake/transactions.cpp

```diff
@@ -78,6 +129,14 @@ StatusOr<TabletMetadataPtr> publish_version(TabletManager* tablet_mgr, int64_t t
         return error;
     };
 
+    int64_t ori_base_version = base_version;
+    adjust_base_version(tablet_id, tablet_mgr, &base_version);
+    if (base_version > new_version) {
```

Comment:
Should judge whether publish mode is batch by txns_size。If convert publish mode from batch to single and retry，the base version may greater than new_version